frontend/dockerfile/builder/build.go

```diff
@@ -829,13 +829,36 @@ func contextByName(ctx context.Context, c client.Client, name string, platform *
 		ref := strings.TrimPrefix(vv[1], "//")
 		imgOpt := []llb.ImageOption{
 			llb.WithCustomName("[context " + name + "] " + ref),
-			llb.WithMetaResolver(c),
 		}
 		if platform != nil {
 			imgOpt = append(imgOpt, llb.Platform(*platform))
 		}
+
+		named, err := reference.ParseNormalizedNamed(ref)
+		if err != nil {
+			return nil, nil, nil, err
+		}
+
+		named = reference.TagNameOnly(named)
+
+		_, data, err := c.ResolveImageConfig(ctx, named.String(), llb.ResolveImageConfigOpt{
```

Comment:
Ideally we would pass the `ResolveMode` and `Log` as in https://github.com/moby/buildkit/blob/master/frontend/dockerfile/dockerfile2llb/convert.go#L336-L337 . Think you can pass `ResolveMode` directly to `contextByNameFunc` like it is done for default platform.

Reply:
Made a change.